### PR TITLE
Ts-19-v2-21-refactor

### PR DESCRIPTION
**Pull request by Akr1s**

### CODE REVIEW

File: frontend/src/components/ClassForm/ClassForm.js [hunk 1]
```diff
@@ -24,37 +24,43 @@ import {
...
-        props.initialize(initialValues);
+        initialize(initialValues);
     }, [classScheduleOne]);
 
     const setEndTime = (startTime) =>
-        props.change(
-            'endTime',
-            moment(startTime, 'HH:mm').add(CLASS_DURATION, 'h').format('HH:mm'),
-        );
+        change('endTime', moment(startTime, 'HH:mm').add(CLASS_DURATION, 'h').format('HH:mm'));
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/ClassForm/ClassForm.js]: 'HH:mm' to constants

File: frontend/src/components/DownloadLink/DownloadLink.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            href={downloadLink || ''}
+            target="_blank"
+            rel="noreferrer noopener"
+            variant="contained"
+            color="primary"
+            className="pdf_link"
+            download
+        >
+            <MdPictureAsPdf className="svg-btn" />
+            {i18n.t('common:download_pdf')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/DownloadLink/DownloadLink.js]: change to constant COMMON_DOWNLOAD_PDF

File: frontend/src/components/GroupCard/GroupCard.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                        title={t(COMMON_SET_ENABLED)}
+                        onClick={() => {
+                            showConfirmDialog(groupItem.id, dialogTypes.SET_VISIBILITY_ENABLED);
+                        }}
+                    />
+                )}
+                <Link to={`${links.GroupList}${links.Group}${links.Delete}/${groupItem.id}`}>
+                    <MdDelete
+                        className="group__buttons-delete link-href"
+                        title={t('delete_title')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/GroupCard/GroupCard.js]: could you change to DELETE_TITLE_LABEL?

File: frontend/src/components/GroupCard/GroupCard.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                        onClick={() => showConfirmDialog(groupItem.id)}
+                    />
+                </Link>
+                <Link to={`${links.GroupList}${links.Group}/${groupItem.id}${links.AddStudent}`}>
+                    <FaUserPlus
+                        title={t('formElements:student_add_label')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/GroupCard/GroupCard.js]: could you change to FORM_STUDENT_ADD_LABEL?

File: frontend/src/components/GroupCard/GroupCard.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                        }}
+                    />
+                </Link>
+            </div>
+            <p className="group-card__description">{`${t(GROUP_LABEL)}:`}</p>
+            <h1 className="group-card__number">{getShortTitle(groupItem.title, 5)}</h1>
+            <Link to={`${links.GroupList}${links.Group}/${groupItem.id}${links.ShowStudents}`}>
+                <span className="students-group">
+                    <FaUsers
+                        title={t('formElements:show_students')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/GroupCard/GroupCard.js]: Could you change to FORM_SHOW_STUDENTS?

File: frontend/src/components/DownloadLink/DownloadLink.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    PUBLIC_DOWNLOAD_GROUP_SCHEDULE_URL,
+    PUBLIC_DOWNLOAD_TEACHER_SCHEDULE_URL,
+} from '../../constants/axios';
+
+const getDownloadLink = (entityId, semesterId, languageToRequest) => ({
+    group: `${PUBLIC_DOWNLOAD_GROUP_SCHEDULE_URL}?groupId=${entityId}&semesterId=${semesterId}${languageToRequest}`,
+    teacher: `${PUBLIC_DOWNLOAD_TEACHER_SCHEDULE_URL}?teacherId=${entityId}&semesterId=${semesterId}${languageToRequest}`,
+});
+
+const DownloadLink = (entity, semesterId, entityId) => {
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/DownloadLink/DownloadLink.js]: Is it right method to destructuring props as parameters of function component?

File: frontend/src/components/GroupSchedulePage/GroupSchedulePage.js [hunk 7]
```diff
@@ -1,297 +1,120 @@
...
-        if (scheduleType === 'group') getGroupSchedule(groupId, semesterId);
-    }, [groupId]);
-    useEffect(() => {
-        if (scheduleType === 'teacher') getTeacherSchedule(teacherId, semesterId);
-    }, [teacherId]);
-
+        getFullSchedule();
+    }, [place]);
     useEffect(() => {
         if (scheduleType === 'full' && fullSchedule.length === 0) {
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/GroupSchedulePage/GroupSchedulePage.js:40]: Could you use isEmpty from Lodash?

File: frontend/src/components/ScheduleItem/ScheduleItem.js [hunk 8]
```diff
@@ -3,15 +3,31 @@ import { MdDelete, MdEdit } from 'react-icons/md';
...
+        translation: t,
+        item: propItem,
+        inBoard,
+        deleteItem,
+        editItem,
+        fStrLetterCapital,
+        addition,
+        className,
+    } = props;
     let { lesson } = props;
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/ScheduleItem/ScheduleItem.js:27]: add to above destructuring

File: frontend/src/components/LessonsList/LessonsList.js [hunk 9]
```diff
@@ -7,25 +7,26 @@ import i18n from 'i18next';
...
+import { COPY_LESSON, DELETE_LESSON, EDIT_LESSON } from '../../constants/translationLabels/common';
 
 const LessonsList = (props) => {
-    const { lessons } = props;
-
-    const t = props.translation;
-
+    const { lessons, translation: t, onCopyLesson, onSelectLesson, onClickOpen } = props;
+    // TODO replece with capitalize
     const firstStringLetterCapitalHandle = (str) => {
```
Comment (Yashchemskyi-Oleksandr) [inline frontend/src/components/LessonsList/LessonsList.js:16]: Can import from helper for less code

File: frontend/src/components/Schedule/Schedule.js [hunk 10]
```diff
@@ -121,7 +129,7 @@ const Schedule = (props) => {
         },
     });
     const elClasses = useStyles();
-
+    // TODO delete this and replace with unified capitalize
     const firstStringLetterCapitalHandle = (str) => {
```
Comment (Yashchemskyi-Oleksandr) [inline frontend/src/components/Schedule/Schedule.js:133]: As above

**Review by Niknikson — APPROVED** [review]

File: frontend/src/components/AddNewRoomType/AddNewRoomType.js [hunk 11]
```diff
@@ -92,7 +94,7 @@ let NewRoomType = (props) => {
                             <span className="buttons">
                                 <FaEdit
                                     className="btn edit"
-                                    onClick={() => handleEdit(roomType.id)}
+                                    onClick={() => getOneNewTypeService(roomType.id)}
```
Comment (andrii-kuzmichov) [inline frontend/src/components/AddNewRoomType/AddNewRoomType.js:97]: clarify name of function

File: frontend/src/components/AddRoomForm/AddRoomForm.js [hunk 12]
```diff
@@ -13,49 +13,58 @@ import Card from '../../share/Card/Card';
...
-                props.initialize({
-                    name: props.oneRoom.name,
-                    type: props.oneRoom.type.id,
-                    id: props.oneRoom.id,
+        if (oneRoom) {
+            if (oneRoom.id) {
+                initialize({
+                    name: oneRoom.name,
+                    type: oneRoom.type.id,
+                    id: oneRoom.id,
```
Comment (andrii-kuzmichov) [inline frontend/src/components/AddRoomForm/AddRoomForm.js:36]: to do destructuring object

File: frontend/src/components/AddSubjectForm/AddSubjectForm.js [hunk 13]
```diff
@@ -10,36 +10,43 @@ import { SUBJECT_FORM } from '../../constants/reduxForms';
...
-        if (props.subject) {
-            if (props.subject.id) {
-                props.initialize({
-                    id: props.subject.id,
-                    name: props.subject.name,
+        if (subject) {
+            if (subject.id) {
+                initialize({
+                    id: subject.id,
+                    name: subject.name,
```
Comment (andrii-kuzmichov) [inline frontend/src/components/AddSubjectForm/AddSubjectForm.js:30]: the same as above and clarify naming

File: frontend/src/components/AddTeacherForm/AddTeacherForm.js [hunk 14]
```diff
@@ -40,15 +53,15 @@ let AddTeacher = (props) => {
         if (teacherId) {
             initializeFormHandler(teacher);
         } else {
-            props.initialize();
+            initialize();
```
Comment (andrii-kuzmichov) [inline frontend/src/components/AddTeacherForm/AddTeacherForm.js:56]: why we need it?

File: frontend/src/components/AddTeacherForm/AddTeacherForm.js [hunk 15]
```diff
@@ -40,15 +53,15 @@ let AddTeacher = (props) => {
...
         }
     }, [teacherId]);
 
     return (
-        <Card class="form-card teacher-form">
+        <Card additionClassName="form-card teacher-form">
             <form className="createTeacherForm w-100" onSubmit={handleSubmit}>
                 <h2 className="form-title">
-                    {teacherId ? t('edit_title') : t('create_title')} {t('teacher_a_label')}
+                    {teacherId ? t(EDIT_TITLE) : t(CREATE_TITLE)} {t(TEACHER_A_LABEL)}
```
Comment (andrii-kuzmichov) [inline frontend/src/components/AddTeacherForm/AddTeacherForm.js:64]: Thinking about unify `<h2>` with similar label

File: frontend/src/components/ClassForm/ClassForm.js [hunk 16]
```diff
@@ -24,37 +24,43 @@ import {
...
-    const { handleSubmit, pristine, onReset, submitting, classScheduleOne } = props;
+    const { handleSubmit, pristine, onReset, submitting, classScheduleOne, initialize, change } =
+        props;
     useEffect(() => {
         let initialValues = {};
         if (classScheduleOne) {
             initialValues = classScheduleOne;
         }
-        props.initialize(initialValues);
+        initialize(initialValues);
```
Comment (andrii-kuzmichov) [inline frontend/src/components/ClassForm/ClassForm.js:47]: replace this code to `initialize(classScheduleOne || {});`

File: frontend/src/components/GroupSchedulePage/GroupSchedulePage.js (hunk 7 above)
Comment (andrii-kuzmichov) [inline frontend/src/components/GroupSchedulePage/GroupSchedulePage.js:40]: let add constants for ScheduleType

File: frontend/src/components/GroupSchedulePageTop/GroupSchedulePageTop.js [hunk 17]
```diff
@@ -37,69 +46,72 @@ const useStyles = makeStyles((theme) => ({
...
-        } else {
-            setGroupDisabled(true);
-        }
-    }, [groups]);
+        showAllPublicTeachersService();
+        showAllPublicSemestersService();
+        initialize({
+            semester: initialValues.semester,
+            group: initialValues.group,
+            teacher: initialValues.teacher,
```
Comment (andrii-kuzmichov) [inline frontend/src/components/GroupSchedulePageTop/GroupSchedulePageTop.js:72]: Destructuring

File: frontend/src/components/GroupSchedulePageTop/GroupSchedulePageTop.js [hunk 18]
```diff
@@ -37,69 +46,72 @@ const useStyles = makeStyles((theme) => ({
...
+        });
+    }, []);
+    useEffect(() => showAllPublicGroupsService(semesterId), [semesterId]);
+    useEffect(() => setGroupDisabled(isEmpty(groups)), [groups]);
+
     const renderSemesterList = () => {
-        if (semesters && semesters.length > 1) {
+        if (!isEmpty(semesters)) {
```
Comment (andrii-kuzmichov) [inline frontend/src/components/GroupSchedulePageTop/GroupSchedulePageTop.js:79]: are you sure that will work correct?

File: frontend/src/components/WeekRoomInfo/WeekRoomInfo.js [hunk 19]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
```
Comment (andrii-kuzmichov) [inline frontend/src/components/WeekRoomInfo/WeekRoomInfo.js:1]: add `TODO` for refactor this file

**Review by andrii-kuzmichov — APPROVED** [review]

**Review by Yashchemskyi-Oleksandr — APPROVED** [review]

**Review by HiroFive — APPROVED** [review]

**Pull request merged**